apps/antalmanac/src/lib/tourExampleGeneration.ts

```diff
@@ -0,0 +1,185 @@
+
+export function addSampleClasses() {
+    if (AppStore.getAddedCourses().length > 0) return;
+
+    const sampleClasses: Array<ScheduleCourse> = [
+        {
+            courseTitle: 'Nice',
+            deptCode: 'GEN&SEX',
+            courseNumber: '69',
+            instructors: ['Your mother'],
+            meetings: [
+                {
+                    bldg: ['DBH'],
+                    days: 'MWF',
+                    startTime: {
+                        hour: 10,
+                        minute: 0,
+                    },
+                    endTime: {
+                        hour: 10,
+                        minute: 50,
+                    },
+                    timeIsTBA: false,
+                },
+            ],
+        },
+        {
+            meetings: [
+                {
+                    bldg: ['ELH 100'],
+                    days: 'TuTh',
+                    startTime: {
+                        hour: 9,
+                        minute: 30,
+                    },
+                    endTime: {
+                        hour: 10,
+                        minute: 50,
+                    },
+                    timeIsTBA: false,
+                },
+            ],
+        },
+    ].map(sampleClassFactory);
+
+    sampleClasses.forEach((sampleClass) => {
+        AppStore.addCourse(sampleClass);
+    });
+}
+
+export function randint(min: number, max: number): number {
+    return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+function randomWeekday(): DayOfWeek {
+    return daysOfWeek[randint(0, 6)];
+}
+
+function randomClasstime(): HourMinute {
+    return {
+        hour: randint(8, 17),
+        minute: randint(0, 5) * 10,
+    };
+}
+
+export function sampleMeetingsFactory({
+    bldg = ['DBH 1200'],
+    days = 'MWF',
+    startTime = {
+        hour: 11,
+        minute: 0,
+    },
+    endTime = {
+        hour: 11,
+        minute: 50,
+    },
+    timeIsTBA = false,
+}: Partial<WebsocSectionMeeting>) {
+    return [
+        {
+            bldg,
+            days,
+            startTime,
+            endTime,
+            timeIsTBA,
+        },
+    ];
+}
+
+export function sampleFinalExamFactory({
+    examStatus = 'SCHEDULED_FINAL',
+    dayOfWeek,
+    month = 12,
+    day = 12,
+    startTime,
+    endTime,
+    bldg = [],
+}: Partial<WebsocSectionFinalExam>): WebsocSectionFinalExam {
+    if (examStatus == 'NO_FINAL')
+        return {
+            examStatus,
+            dayOfWeek: 'Mon',
+            month: 0,
+            day: 0,
+            startTime: {
+                hour: 0,
+                minute: 0,
+            },
+            endTime: {
+                hour: 0,
+                minute: 0,
+            },
+            bldg,
+        };
+
+    return {
+        examStatus,
+        dayOfWeek: dayOfWeek ?? randomWeekday(),
+        month,
+        day,
+        startTime: startTime ?? randomClasstime(),
+        endTime: endTime ?? randomClasstime(),
```

Comment:
Not sure if this should be random; Is there an advantage over have a specific fallback time?

Reply:
There isn't. I just didn't want to spell them out.